pkg/prometheus/rules.go

```diff
@@ -221,11 +224,20 @@ func (c *Operator) selectRules(p *monitoringv1.Prometheus, namespaces []string)
 	return rules, nil
 }
 
-func generateContent(promRule monitoringv1.PrometheusRuleSpec) (string, error) {
+func generateContent(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
 	content, err := yaml.Marshal(promRule)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to marshal content")
 	}
+	_, errs := rulefmt.Parse(content)
```

Comment:
What will happen if rule has additional fields like `partial_response_strategy: "warn"` needed sometimes by thanos ([docs](https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md#partial-response))? Can we add a test case for this?

Reply:
@paulfantom - good call!
You're right - partial_response_strategy field is indeed defined in https://github.com/prometheus-operator/prometheus-operator/blob/fb24ca8fd26597c5a2d69ccac949aa5a76cf29ce/pkg/apis/monitoring/v1/types.go#L1328

but not in upstream prometheus 

https://github.com/prometheus/prometheus/blob/b1ed4a0a663d0c62526312311c7529471abbc565/pkg/rulefmt/rulefmt.go#L107

The latter is used for validation so we need to find a way to reconcile this.

Btw - this may be considered a bug in the current version as the webhook validation does the same thing so when enabled, valid thanos rules may fail validation

Reply:
@paulfantom can you check the latest version of this PR - I extended the check to ignore `PartialResponseStrategy` field if defined

Reply:
> Btw - this may be considered a bug in the current version as the webhook validation does the same thing so when enabled, valid thanos rules may fail validation

Definitely a bug, could you open an issue for it (or maybe you would you like to work on this?)? It might be good to reuse code from this PR in the admission webhook part too. Ideally, such a fix would be in a separate PR though.

Reply:
Agree - let me wrap up this PR and file a separate issues and PR for the admission hook